### DISCUSSION TITLE
Android remoteMessage getData() possible?

### DISCUSSION THREAD

**Discussion by LNRS-iampnelson** [invertase/react-native-firebase#7432] [Q&A]
Hello,

Is there a way to pass the received notification + data from JavaScript back to native Java to where I can leverage remoteMessage.getData()?

I would like to do something to the similar from the native Java side.

`
// Check if message contains a data payload.
    if (remoteMessage.getData().size() > 0) {
        Log.d(TAG, "Message data payload: " + remoteMessage.getData());

        if (/* Check if data needs to be processed by long running job */ true) {
            // For long-running tasks (10 seconds or more) use WorkManager.
            scheduleJob();
        } else {
            // Handle message within 10 seconds
            handleNow();
        }

    }
`

I can successfully pass the payload received from JavaScript to native via a module I created. But, how can I make the payload be the equivalent of remoteMessage. Is this possible?

Thank you!

**Comment by mikehardy** [comment]
I was going to suggest a custom native module, looks like you are on the right track there.
Not sure why the data you send is not the data you want though ?

> But, how can I make the payload be the equivalent of remoteMessage

The API you make in your custom module may have any signature you want, I suggest sending the map data type through which should allow the remote message JSON to fit through it, as I believe they have the same "shape"

Alternatively, you could JSON.toString it and send it as a string ?
